### ISSUE TITLE
Is it possible to use enum at java side, but without enum at db side?

### ISSUE THREAD

**Issue by plandem** [EvidentSolutions/dalesbred#12]
i have such model:

``` java
final public class Constant {
    public Integer itemid;
    public ItemType itemtype;
    public ConstantType constanttype;
    public Float constantvalue;
    public String constanttext;
}
```

where itemType is:

``` java
public enum ItemType {
    folder(60),
    device(99);

    private final Integer type;

    ItemType(@NotNull Integer type) {
        this.type = type;
    }

    @NotNull
    public Integer getType() {
        return this.type;
    }
}
```

and ConstantType is:

``` java
public enum ConstantType {
    longitude(2150),
    latitude(2151),
    no(34),
    serial(8),
    lastName(50),
    firstName(51),
    middleName(52),
    factory(21);

    private final Integer type;

    ConstantType(@NotNull Integer type) {
        this.type = type;
    }

    @NotNull
    public Integer getType() {
        return this.type;
    }
}
```

but when i try to fetch results, i get next error:

```
java.lang.IllegalArgumentException: No enum constant com.luxms.app.model.ItemType.60
    at java.lang.Enum.valueOf(Enum.java:238)
    at org.dalesbred.dialect.Dialect.parseNativeDatabaseEnum(Dialect.java:104)
    at org.dalesbred.dialect.Dialect.parseDatabaseEnum(Dialect.java:91)
```

**Comment by plandem** [comment]
now i have to implement constructor and resolvers for enums, but it's so annoying.

``` java
    /**
     * Resolve enum via value
     * @param value value to get enum
     * @return ConstantType
     */
    @NotNull
    public static ConstantType getEnum(@NotNull Integer value) {
        for(ConstantType type: values()) {
            if(type.getType().equals(value)) {
                return type;
            }
        }

        throw new IllegalArgumentException("No enum constant for value " + value);
    }
```

and 

``` java
    public Constant(Integer itemid, Integer itemtype, Integer constanttype, Float constantvalue, String constanttext) {
        this.itemid = itemid;
        this.itemtype = ItemType.getEnum(itemtype);
        this.constanttype = ConstantType.getEnum(constanttype);
        this.constanttext = constanttext;
        this.constantvalue = constantvalue;
    }
```

**Comment by komu** [comment]
Yes, you can register custom type-conversions with `TypeConversionRegistry`. Since in your case you already have functions for converting `Integer` to `ConstantType` and back, you can register those directly:

``` java
TypeConversionRegistry registry = db.getTypeConversionRegistry();
registry.registerConversions(Integer.class, ConstantType.class,
    ConstantType::getEnum, ConstantType::getType);
```

**Comment by komu** [comment]
I just released version 1.0.0-rc.1 with a simplified way to register your enum conversion:

``` java
TypeConversionRegistry registry = db.getTypeConversionRegistry();
registry.registerConversions(ConstantType.class, ConstantType::getType);
```

When saving the value, this will call `ConstantType::getType` for the value to look up the value to be saved. When loading the value, it will iterate over all enum constants of the class, call `ConstantType::getType` for each of them and return the matching constant. So basically the stuff that your `getEnum` does now comes for free for any enum.
